### PR TITLE
Fix off by one error in parallel scheduler

### PR DESCRIPTION
**Pull request by aludvik**
Signed-off-by: Adam Ludvik <ludvik@bitwise.io>

### CODE REVIEW

**Comment by aludvik** [comment]
This closes STL-1055

File: validator/tests/test_scheduler/test_schedulers_with_yaml.py [hunk 1]
```diff
@@ -285,6 +285,23 @@ def test_complex_dependency_failures(self):
             context_manager=context_manager,
             name='complex_dependency_failures.yaml')
 
+    def test_single_txn_per_batch(self):
```
Comment (peterschwarz) [inline validator/tests/test_scheduler/test_schedulers_with_yaml.py]: How does this test prevent a regression for this issue?  It's not at all clear from the test case itself. 

Can you speak a bit more about these items?

File: validator/tests/test_scheduler/test_schedulers_with_yaml.py [hunk 2]
```diff
@@ -285,6 +285,23 @@ def test_complex_dependency_failures(self):
...
+        """Tests the schedulers against the
+        test_scheduler/data/long_chain_of_inputs_outputs.yaml file.
```
Comment (peterschwarz) [inline validator/tests/test_scheduler/test_schedulers_with_yaml.py]: Missing this file.

File: validator/tests/test_scheduler/test_schedulers_with_yaml.py (hunk 1 above)
Comment (aludvik) [reply validator/tests/test_scheduler/test_schedulers_with_yaml.py]: Yeah, sorry the explanation was in the file I forgot to commit...

**Comment by aludvik** [comment]
This fix is currently running on PSIM-LR1

**Pull request merged**